blockstack/blockstack-browser
# Feature/gaia association token

This PR adds support for generating and passing along a Gaia association token on sign-in.  This will allow users who onboarded with a private Gaia hub to use that Gaia hub without having to whitelist individual app addresses.

## How To Test

* Start up your own Gaia hub, but do _not_ whitelist any addresses yet.  This is necessary for onboarding to work.
* Make a new Blockstack ID and onboard with your custom Gaia hub.
* Edit your Gaia hub's config to whitelist just your ID-address, and restart it.
* Proceed to sign in to an app with your new Blockstack ID.
* Confirm that you are able to use apps normally -- your Gaia hub should be receiving and storing data, and the auth header should contain an association token.
* Confirm that your Blockstack ID's zone file points to your custom Gaia hub, and that your profile's `apps` point to your custom Gaia hub.

You can see the results in my test Blockstack ID `testlocalgaiahub.test-personal.id`.
